### DISCUSSION TITLE
Conversion error: multi-page pdf to multiple single page png files, color unavailable

### DISCUSSION THREAD

**Discussion by rajibando** [ImageMagick/ImageMagick6#252] [General]
### ImageMagick version

8:6.9.11.60+dfsg-1.3+deb11u1

### Operating system, version and so on

Official Debian GNU/Linux Live 11.6.0 lxde 2022-12-17T11:46

### Description

Want to convert a multi-page text-only, formatted file, `textfile.pdf` into multiple page png files with Dark Blue texts. Used the following code (initial code line courtesy rich2005 of gimp-forum dot net):
`convert -density 100 -colorspace sRGB -background none -fill 'rgb(0,0,139)' -quality 05 textfile.pdf imagecon/imagefile%03d.png`
Output:
> convert-im6.q16: profile 'icc': 'RGB ': RGB color space not permitted on grayscale PNG `imagecon/imagefile%03d.png' @ warning/png.c/MagickPNGWarningHandler/1668.

The png files that are produced are all with black pixels of characters.

One such png file, when opened with gimp, "GIMP Message" reports:

> Execution error for procedure 'gimp-image-set-color-profile':
> ICC profile validation failed: Color profile is not for grayscale color space

How to solve this issue and get output image files as desired?

Thanks and best wishes.

**Comment by urban-warrior** [comment]
Post a link to `textfile.pdf` so we can inspect it and determine the source of the issue.

**Comment by rajibando** [comment]
This link of the directory in the Google Drive, along with the
concerned pdf file, the code-line and output error in the text file
OutputError.txt, is here:
https://drive.google.com/drive/folders/1eCa0OlXm-7g3Jlt4NYLmh0oiQlIGHjus

".png" files page-wise for all pages are produced, but in Black and White.

One sample converted file in the imagecon sub-directory

Best wishes,
Rajibando

---------- Received message ----------
From: Cristy ***@***.***>
Date: Tue, 04 Jul 2023 04:43:55 -0700
Subject: Re: [ImageMagick/ImageMagick] Conversion error: multi-page
pdf to multiple single page png files, color unavailable (Discussion
ImageMagick/ImageMagick#6457)
To: ImageMagick/ImageMagick ***@***.***>
Cc: rajibando ***@***.***>, Author ***@***.***>

Post a link to `textfile.pdf` so we can inspect it and determine the
source of the issue.

-- 
Reply to this email directly or view it on GitHub:
https://github.com/ImageMagick/ImageMagick/discussions/6457#discussioncomment-6352776

**Comment by urban-warrior** [comment]
Try this command:

```
convert -density 400 textfile.pdf -fill 'rgb(0,0,139)' -opaque black -resize 25% imagefile%03d.png
```

**Comment by rajibando** [comment]
Dear All concerned,

Mr. Bob Friesenhahn, maintainer, graphicsmagick, has solved my query,
suggesting two lines of codes, either of which worked as intended, and
I have thanked him, in my post here:
https://sourceforge.net/p/graphicsmagick/mailman/message/37865936/

The complete lines are as follows:
convert -density 100 LoopEternal.pdf -colorspace Gray -colorspace rgb \
-fill blue -colorize 0/0/100 -transparent white +adjoin \
imagecon/image-%03d.png

and

convert -density 100 LoopEternal.pdf -colorspace Gray -colorspace rgb \
-fill blue -colorize 0/0/100 -fuzz 1% -transparent white +adjoin \
imagecon/name-%03d.png

They work as intended.

I have requested his advice on implementing my colour choice:
The identifier for Dark Blue:
PMS: 3591 C
Hex Color: #06038D;
RGB: (3,37,126)
CMYK: (100,87,0,25)

I am sharing the lines for common empowerment.

Best wishes,
Rajib
Etc.

**Reply by fmw42** [reply]
You should be using -colorspace sRGB in Imagemagick not -colorspace RGB

**Comment by rajibando** [comment]
I am extremely sorry and profusely apologise for posting my reply
while you sent yours. Your line works as intended. With the colour
requested.

So thank you indeed, Mr. Cristy, and best wishes,

Rajibando
Etc.



---------- Received message ----------
From: Cristy ***@***.***>
Date: Tue, 04 Jul 2023 07:27:14 -0700
Subject: Re: [ImageMagick/ImageMagick] Conversion error: multi-page
pdf to multiple single page png files, color unavailable (Discussion
ImageMagick/ImageMagick#6457)
To: ImageMagick/ImageMagick ***@***.***>
Cc: rajibando ***@***.***>, Author ***@***.***>

Try this command:

```
convert -density 400 textfile.pdf -fill 'rgb(0,0,139)' -opaque black
-resize 25% imagefile%03d.png
```
[   ...   ]
